File: find-pair.py
def find_pair(n, nums):
    seen = set()
    for num in nums:
        target = n - num
        if target in seen:
            return str(num) + "\n" + str(target)
        seen.add(num)


# Read input from user
n = int(input())
nums = list(map(int, input().split()))
print(find_pair(n, nums))

File: find-pair.cpp
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

pair<int, int> findPair(const vector<int>& nums, int target) {
    unordered_map<int, int> numMap;
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        if (numMap.find(complement) != numMap.end()) {
            return {complement, nums[i]};
        }
        numMap[nums[i]] = i;
    }
    return {};
}

int main() {
    int n;
    cin >> n;
    vector<int> numbers(n);
    for (int i = 0; i < n; i++) {
        cin >> numbers[i];
    }
    int target;
    cin >> target;
    pair<int, int> result = findPair(numbers, target);
    if (!result.first || !result.second) {
        cout << "No such pair found.\n";
    } else {
        cout << result.first << '\n' << result.second << '\n';
    }
    return 0;
}